javalib/src/main/scala/java/lang/Double.scala

```diff
@@ -48,30 +59,132 @@ object Double {
   final val MIN_EXPONENT      = -1022
   final val SIZE              = 64
 
-  @inline def valueOf(doubleValue: scala.Double): Double =
-    new Double(doubleValue)
+  @inline def compare(x: scala.Double, y: scala.Double): scala.Int =
+    if (x > y) 1
+    else if (x > y) -1
+    else if (x == y && 0.0d != x) 0
+    else {
+      if (isNaN(x)) {
+        if (isNaN(y)) 0
+        else 1
+      } else if (isNaN(y)) {
+        -1
+      } else {
+        val d1 = doubleToRawLongBits(x)
+        val d2 = doubleToRawLongBits(y)
+        ((d1 >> 63) - (d2 >> 63)).toInt
+      }
+    }
 
-  @inline def valueOf(s: String): Double = valueOf(parseDouble(s))
+  @inline def doubleToLongBits(value: scala.Double): scala.Long =
+    if (value != value) 0x7ff8000000000000L
+    else doubleToRawLongBits(value)
 
-  def parseDouble(s: String): scala.Double = ???
+  @native def doubleToRawLongBits(value: scala.Double): scala.Long
 
-  @inline def toString(d: scala.Double): String =
-    "" + d
+  @inline def hashCode(value: scala.Double): scala.Int = {
+    val v = doubleToLongBits(value)
+    (v ^ (v >>> 32)).toInt
+  }
+
+//  def isFinite(d: scala.Double): scala.Boolean = ???
 
-  def compare(a: scala.Double, b: scala.Double): scala.Int = ???
+  @inline def isInfinite(v: scala.Double): scala.Boolean =
+    v == POSITIVE_INFINITY || v == NEGATIVE_INFINITY
 
   @inline def isNaN(v: scala.Double): scala.Boolean =
     v != v
 
-  @inline def isInfinite(v: scala.Double): scala.Boolean =
-    v == POSITIVE_INFINITY || v == NEGATIVE_INFINITY
+  @native def longBitsToDouble(bits: scala.Long): scala.Double
+
+  def max(a: scala.Double, b: scala.Double): scala.Double =
+    Math.max(a, b)
+
+  def min(a: scala.Double, b: scala.Double): scala.Double =
+    Math.min(a, b)
+
+  @inline def parseDouble(s: _String): scala.Double =
+    NumberParser.parseDouble(s)
+
+  def sum(a: scala.Double, b: scala.Double): scala.Double =
+    a + b
+
+  def toHexString(d: scala.Double): _String = {
+    if (d != d) {
+      "NaN"
+    } else if (d == POSITIVE_INFINITY) {
+      "Infinity"
+    } else if (d == NEGATIVE_INFINITY) {
+      "-Infinity"
+    } else {
+      val bitValue = doubleToLongBits(d)
+      val negative = (bitValue & 0x8000000000000000L) != 0
+      val exponent = (bitValue & 0x7FF0000000000000L) >>> 52
+      var significand = bitValue & 0x000FFFFFFFFFFFFFL
+      if (exponent == 0 && significand == 0) {
+        if (negative) "-0x0.0p0"
+        else "0x0.0p0"
+      } else {
+        val hexString = new java.lang.StringBuilder(10)
```

Comment:
The longest possible string has length 24, unless I'm mistaken. It is `-0x0.fffffffffffffp-1022`, which would be produced by `(-Double.MinPositiveValue).toHexString()`. Hence, you should create the `StringBuilder` with an argument of 24 instead of 10.